src/v1_0/rws_client.cpp

```diff
@@ -250,20 +250,37 @@ std::string RWSClient::generateFilePath(const FileResource& resource)
   return Services::FILESERVICE + "/" + resource.directory + "/" + resource.filename;
 }
 
+// Try finding xml tag with class="code" attribute containing retcode. Does not throw.
+std::optional<int> tryParseRetcode(std::string const& content) noexcept
+{
+  std::optional<int> result;
+  try
+  {
+    auto xml = parseXml(content);
+    std::string code_str = xmlFindTextContent(xml, XMLAttribute{"class", "code"});
+    if (!code_str.empty())
+        result = std::stoi(code_str);
+  } catch (...) {}
```

Comment:
This is of course shorter than catching `[std::invalid_argument](https://en.cppreference.com/w/cpp/error/invalid_argument)` and `[std::out_of_range](https://en.cppreference.com/w/cpp/error/out_of_range)`. But I believe it is better to be more explicit on the types of exceptions that you expect, and not try to catch what you don't expect or can't handle.